content/courses/advanced-ada/parts/resource_management/access_types.rst

```diff
@@ -15,29 +15,409 @@ Access types: Terminology
+:ada:`new Integer` allocates a new integer object in that pool. Therefore,
+access types of this kind are called pool-specific access types. (We discuss
+:ref:`more about these types <Adv_Ada_Pool_Specific_Access_Types>` later.)
+
+It is also possible to access objects that were allocated on the stack. To do
+that, however, we cannot use pool-specific access types because |mdash| as the
+name suggests |mdash| they're only allowed to access objects that were
+allocated in the specific pool associated with the type. Instead, we have to
+use general access types in this case:
+
+.. code:: ada run_button project=Courses.Advanced_Ada.Resource_Management.Access_Types.Access_Types_Allocation.General_Access_Types
+
+    procedure Show_General_Access_Type is
+
+       --  Declaring general access type:
+       type Integer_Access is access all Integer;
+
+       --  Declaring access object:
+       A1 : Integer_Access;
+
+       --  Allocating an integer variable on the
+       --  stack:
+       I : aliased Integer;
+
+    begin
+       --  Getting access to an integer variable that
+       --  was allocated on the stack
+       A1 := I'Access;
+
+    end Show_General_Access_Type;
+
+In this example, we declare the general access type :ada:`Integer_Access` and
+the access object :ada:`A1`. To initialize :ada:`A1`, we write :ada:`I'Access`
+to get access to an integer variable :ada:`I` that was allocated on the stack.
+(For the moment, don't worry much about these details: we'll talk about general
+access types again when we introduce the topic of
+:ref:`aliased objects <Adv_Ada_Aliased_Objects>` later on.)
+
+.. admonition:: For further reading...
+
+    Note that it is possible to use general access types to allocate objects on
+    the heap:
+
+    .. code:: ada run_button project=Courses.Advanced_Ada.Resource_Management.Access_Types.Access_Types_Allocation.General_Access_Types_Heap
+
+        procedure Show_Simple_Allocation is
+
+           --  Declaring general access type:
+           type Integer_Access is access all Integer;
+
+           --  Declaring access object:
+           A1 : Integer_Access;
+
+        begin
+           --
+           --  Allocating an integer variable on the heap
+           --  and initializing an access object of
+           --  the general access type Integer_Access.
+           --
+           A1 := new Integer;
+
+        end Show_Simple_Allocation;
+
+    Here, we're using a general access type :ada:`Integer_Access`, but
+    allocating an integer variable on the heap.
+
+.. admonition:: Important
+
+    In many code examples, we have used the :ada:`Integer` type as the
+    designated subtype of the access types |mdash| by writing
+    :ada:`access Integer`. Although we have used this specific scalar type,
+    we aren't really limited to those types. In fact, we can use *any type* as
+    the designated subtype, including user-defined types, composite types,
+    task types and protected types.
```

Comment:
That's a good argument for using "object" instead of "variable", eg that a protected object is not a variable.